src/set_theory/cardinal/finite.lean

```diff
@@ -34,6 +34,49 @@ lemma card_eq_fintype_card [fintype α] : nat.card α = fintype.card α := mk_to
 @[simp]
 lemma card_eq_zero_of_infinite [infinite α] : nat.card α = 0 := mk_to_nat_of_infinite
 
+lemma card_congr (f : α ≃ β) : nat.card α = nat.card β :=
+cardinal.to_nat_congr f
+
+lemma card_eq_of_bijective (f : α → β) (hf : function.bijective f) : nat.card α = nat.card β :=
+card_congr (equiv.of_bijective f hf)
+
+lemma card_eq_of_equiv_fin {α : Type*} {n : ℕ}
+  (f : α ≃ fin n) : nat.card α = n :=
+by simpa using card_congr f
+
+/-- If the cardinality is positive, that means it is a finite type, so there is
+an equivalence between `α` and `fin (nat.card α)`. See also `finite.equiv_fin`. -/
+def equiv_fin_of_card_pos {α : Type*} (h : nat.card α ≠ 0) :
+  α ≃ fin (nat.card α) :=
+begin
+  casesI fintype_or_infinite α,
+  { simpa using fintype.equiv_fin α },
+  { simpa using h },
+end
+
+lemma card_of_subsingleton (a : α) [subsingleton α] : nat.card α = 1 :=
+begin
+  rw [card_eq_fintype_card],
+  convert fintype.card_of_subsingleton a,
+end
+
+@[simp] lemma card_unique [unique α] : nat.card α = 1 :=
+card_of_subsingleton default
+
+lemma card_eq_one_iff_nonempty_unique : nat.card α = 1 ↔ subsingleton α ∧ nonempty α :=
```

Comment:
```suggestion
lemma card_eq_one_iff_unique : nat.card α = 1 ↔ subsingleton α ∧ nonempty α :=
```
Though I would name all these names `iff_subsingleton_nonempty`.

Reply:
I think `subsingleton_nonempty` would be easier to find than `unique` but it's probably better to stick with the current convention (at least until someone changes them all).